applicationset/controllers/applicationset_controller.go

```diff
@@ -854,10 +854,10 @@ func (r *ApplicationSetReconciler) removeOwnerReferencesOnDeleteAppSet(ctx conte
 	return nil
 }
 
-func (r *ApplicationSetReconciler) performProgressiveSyncs(ctx context.Context, logCtx *log.Entry, appset argov1alpha1.ApplicationSet, applications []argov1alpha1.Application, desiredApplications []argov1alpha1.Application, appMap map[string]argov1alpha1.Application) (map[string]bool, error) {
-	appDependencyList, appStepMap := r.buildAppDependencyList(logCtx, appset, desiredApplications)
+func (r *ApplicationSetReconciler) performProgressiveSyncs(ctx context.Context, logCtx *log.Entry, appset *argov1alpha1.ApplicationSet, applications []argov1alpha1.Application, desiredApplications []argov1alpha1.Application, appMap map[string]argov1alpha1.Application) (map[string]bool, error) {
+	appDependencyList, appStepMap := r.buildAppDependencyList(logCtx, *appset, desiredApplications)
```

Comment:
Why is a pointer required here?

Reply:
I am asking why `buildAppDependencyList` requires a pointer. It doesn't change the applicationset at all.

Reply:
Sorry, I misread! I see you are deferencing it.